run/CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ target_link_libraries(allsim
                                 O2::SimSetup
                                 FairMQ::FairMQ
                                 O2::CPVSimulation
+                                O2::A3DetectorsPassive
```

Comment:
Duplicating the code here seems risky (divergence)  to me, what about defining a list with the additionally linked libraries when upgrades are enabled and including this in the one call to `target_link_libraries`?

Reply:
In any case, linking against this needs to be protected for ENABLE_UPGRADES=ON/OFF?

Reply:
It has indeed to be guarded as we don't want any upgrades-related product to be present in Run3 builds.

Reply:
Sure, it needs to be guarded. But can't we just add one line to the list instead of duplicating the whole construction (which means all future adjustments need to be done in two places)?
```cmake
$<$<BOOL:${ENABLE_UPGRADES}>:O2::A3DetectorsPassive>
```

Reply:
looks definitely better, I'll try it